lib/Differentiator/BaseForwardModeVisitor.cpp

```diff
@@ -1315,7 +1318,11 @@ StmtDiff BaseForwardModeVisitor::VisitUnaryOperator(const UnaryOperator* UnOp) {
            opKind == UnaryOperatorKind::UO_Imag) {
     return StmtDiff(op, BuildOp(opKind, diff.getExpr_dx()));
   } else if (opKind == UnaryOperatorKind::UO_Deref) {
-    return StmtDiff(op, BuildOp(opKind, diff.getExpr_dx()));
+    Expr* dx = diff.getExpr_dx();
+    if (dx == nullptr)
+      return StmtDiff(op, ConstantFolder::synthesizeLiteral(
+                              m_Context.IntTy, m_Context, /*val=*/0));
+    return StmtDiff(op, BuildOp(opKind, dx));
```

Comment:
```suggestion

    if (Expr* dx = diff.getExpr_dx())
      return StmtDiff(op, BuildOp(opKind, dx));
    return StmtDiff(op, ConstantFolder::synthesizeLiteral(
                              m_Context.IntTy, m_Context, /*val=*/0));
```
I know that we are not using this consistently but `IntTy` here is probably not what we want semantically. Shall we give it a try with a `VoidPtrTy` or a `NullPtrTy`?